server/plugin/plugin.go

```diff
@@ -22,88 +17,68 @@ import (
 	"github.com/mattermost/mattermost-plugin-solar-lottery/server/api"
 	"github.com/mattermost/mattermost-plugin-solar-lottery/server/command"
 	"github.com/mattermost/mattermost-plugin-solar-lottery/server/config"
-	sl "github.com/mattermost/mattermost-plugin-solar-lottery/server/solarlottery"
-	"github.com/mattermost/mattermost-plugin-solar-lottery/server/solarlottery/autofill/queue"
-	"github.com/mattermost/mattermost-plugin-solar-lottery/server/solarlottery/autofill/solarlottery"
-	"github.com/mattermost/mattermost-plugin-solar-lottery/server/store"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/constants"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/sl"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/sl/filler/solarlottery"
 	"github.com/mattermost/mattermost-plugin-solar-lottery/server/utils/bot"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/utils/kvstore"
+	"github.com/mattermost/mattermost-plugin-solar-lottery/server/utils/types"
 )
 
 type Plugin struct {
 	plugin.MattermostPlugin
-	configLock *sync.RWMutex
-	config     *config.Config
 
-	httpHandler *api.Handler
-	// notificationHandler sl.NotificationHandler
+	bot    bot.Bot
+	sl     sl.Service
+	api    *api.Service
+	config config.Service
 
-	Templates map[string]*template.Template
+	botUserID string
```

Comment:
There are some issues because `p.botUserID` is cached once, in `OnActivate` but the bot is currently not a "service" and is freely instantiated in a few places. 1/5 suggest leaving as is, but refactoring later, making bot a "service".